pkgs/build-support/appimage/default.nix

```diff
@@ -55,12 +63,33 @@ rec {
       meta = {
         sourceProvenance = with lib.sourceTypes; [ binaryNativeCode ];
       } // meta;
-    } // (removeAttrs args ([ "pname" "version" ] ++ (builtins.attrNames (builtins.functionArgs wrapAppImage)))));
-
-  wrapType2 = args@{ name ? "${args.pname}-${args.version}", src, extraPkgs ? pkgs: [ ], ... }: wrapAppImage
-    (args // {
+    } // (removeAttrs args ([ "pname" "version" ] ++ (lib.attrNames (builtins.functionArgs wrapAppImage)))));
+
+  wrapType2 =
+    { name ? "${args.pname}-${args.version}"
+    , src
+    , extraPkgs ? pkgs: [ ]
+    , extraInstall ? extracted: ""
+    , ...
+    } @ args:
+    let
+      extracted = extract { inherit name src; };
+      extraInstallString = extraInstall extracted;
+    in
+    wrapAppImage (args // {
       inherit name extraPkgs;
-      src = extract { inherit name src; };
+      src = extracted;
+
+      extraInstallCommands = args.extraInstallCommands or ''
+        # Dont include the version number in the binary name
+        if [ -f "$out/bin/${name}" ]; then
+          mv "$out/bin/${name}" "$out/bin/${lib.getName name}"
```

Comment:
name is just here for compatibility reasons because pname + version support is not that old yet. I would prefer to remove it, too and only rely on pname + version

Reply:
I do think having support for just `name` is valuable, a project might use `wrapType2` in a flake or something where an explicit version is not really useful. 

I believe the issue here is that `buildFhsUserEnv` only takes `name` instead of `pname` and `version`, which it bases the binary name off of. Using `name = lib.getName name` in the call to it (see `wrapAppImage`) breaks almost all existing derivations since they manually move the binary to a non-versioned path. That is why I opted to use check for it in bash only if `extraInstallCommands` isn't set.